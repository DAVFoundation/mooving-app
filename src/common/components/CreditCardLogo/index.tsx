import React from 'react';
import { Image, View } from 'react-native';

interface ICreditCardLogoProps {
  brand: string;
  style?: any;
}

const brands = {
  americanexpress: 'https://files.readme.io/97e7acc-Amex.png',
  cartebleau: 'https://files.readme.io/5da1081-cb.png',
  diners: 'https://files.readme.io/8c73810-Diners_Club.png',
  discover: 'https://files.readme.io/caea86d-Discover.png',
  jcb: 'https://files.readme.io/e076aed-JCB.png',
  maestrouk: 'https://files.readme.io/daeabbd-Maestro.png',
  mastercard: 'https://files.readme.io/5b7b3de-Mastercard.png',
  solo: 'https://bluesnap.com/services/hosted-payment-fields/cc-types/solo.png',
  visa: 'https://files.readme.io/9018c4f-Visa.png',
} as any;

export default class CreditCardLogo extends React.Component<ICreditCardLogoProps> {

  public render() {
    const formattedBrandName = this.props.brand.toLowerCase().replace(/\s/g, '');
    const uri = brands[formattedBrandName] || 'https://files.readme.io/d1a25b4-generic-card.png';
    return (
      <View style={this.props.style}>
        <Image source={{uri}} style={{width: 32, height: 24, borderRadius: 2}}/>
      </View>
    );
  }
}
