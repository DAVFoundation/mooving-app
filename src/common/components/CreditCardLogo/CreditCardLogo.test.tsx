import React from 'react';
import renderer from 'react-test-renderer';
import CreditCardLogo from '../CreditCardLogo';

it('renders correctly with defaults', () => {
  const view = renderer
    .create(<CreditCardLogo brand='not-a-real-brand'/>)
    .toJSON();
  expect(view).toMatchSnapshot();
});
