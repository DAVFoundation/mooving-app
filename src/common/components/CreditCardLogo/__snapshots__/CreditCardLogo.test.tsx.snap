// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View>
  <Image
    source={
      Object {
        "uri": "https://files.readme.io/d1a25b4-generic-card.png",
      }
    }
    style={
      Object {
        "borderRadius": 2,
        "height": 24,
        "width": 32,
      }
    }
  />
</View>
`;
